File: backend/dtos/src/main/java/io/mateu/dtos/Badge.java
package io.mateu.dtos;

/**
 * A badge
 *
 * @param theme Theme to be used: info, warning, success, ...
 * @param message The message text to be shown
 * @param icon The icon to be shown
 * @param badgeStyle The badge style: round, square
 * @param iconPosition The icon position: left or right
 */
public record Badge(
        BadgeTheme theme,
        String message,
        String icon,
        BadgeStyle badgeStyle,
        BadgeIconPosition iconPosition
) {

}

File: frontend/src/mateu/shared/apiClients/dtos/Badge.ts
import {BadgeTheme} from "./BadgeTheme";
import {BadgeStyle} from "./BadgeStyle";
import {BadgeIconPosition} from "./BadgeIconPosition";

export default interface Badge {

    theme: BadgeTheme
    message: string | undefined
    icon: string
    badgeStyle: BadgeStyle
    iconPosition: BadgeIconPosition

}
